lib/hook/consumers/github.ex

```diff
@@ -0,0 +1,140 @@
+defmodule Accent.Hook.Consumers.GitHub do
+  @moduledoc """
+  From a project’s integration association and a webhook event from GitHub,
+  sync and add translation as if the operations were made manually by a user.
+
+  """
+  use Accent.Hook.EventConsumer, subscribe_to: [Accent.Hook.Producers.GitHub]
+
+  alias Accent.{Document, Repo, Version}
+  alias Accent.Hook.Consumers.GitHub.AddTranslations
+  alias Accent.Hook.Consumers.GitHub.Sync
+  alias Accent.Hook.Context
+  alias Accent.Plugs.MovementContextParser
+  alias Accent.Scopes.Document, as: DocumentScope
+  alias Accent.Scopes.Version, as: VersionScope
+
+  def handle_events(events, _from, state) do
+    Enum.each(events, &handle_event/1)
+
+    {:noreply, [], state}
+  end
+
+  defp handle_event(%Context{user: user, project: project, event: "push", payload: payload}) do
+    payload[:ref]
+    |> ref_to_version(payload[:default_ref], project)
+    |> sync_and_add_translations(project, user, payload)
+  end
+
+  defp sync_and_add_translations({:ok, version}, project, user, payload) do
+    repo = payload[:repository]
+    token = payload[:token]
+
+    ref = (version && version.tag) || payload[:default_ref]
+    configs = fetch_config(repo, token, ref)
+    trees = fetch_trees(repo, token, ref)
+
+    revisions =
+      project
+      |> Ecto.assoc(:target_revisions)
+      |> Repo.all()
+      |> Repo.preload(:language)
+
+    Sync.persist(trees, configs, project, user, payload, version)
+
+    Enum.each(revisions, fn revision ->
+      AddTranslations.persist(trees, configs, project, user, revision, payload, version)
+    end)
+  end
+
+  defp sync_and_add_translations(_, _, _, _), do: :ok
+
+  def filter_by_patterns(patterns, files) do
+    Enum.group_by(files, fn file -> Enum.find(patterns, &ExMinimatch.match(&1["matcher"], file["path"])) end)
+  end
+
+  def movement_document(project, path) do
+    path =
+      path
+      |> Path.basename()
+      |> MovementContextParser.extract_path_from_filename()
+
+    Document
+    |> DocumentScope.from_path(path)
+    |> DocumentScope.from_project(project.id)
+    |> Repo.one()
+    |> Kernel.||(%Document{project_id: project.id, path: path})
+  end
+
+  def fetch_content(path, token) do
+    with {:ok, %{body: %{"content" => content}}} <- file_server().get(path, headers(token)),
+         decoded_contents <-
+           content
+           |> String.split("\n")
+           |> Enum.reject(&(&1 === ""))
+           |> Enum.map(&Base.decode64/1),
+         true <- Enum.all?(decoded_contents, &match?({:ok, _}, &1)),
+         decoded_content <-
+           decoded_contents
+           |> Enum.map(&elem(&1, 1))
+           |> Enum.join("") do
+      {:ok, decoded_content}
+    else
+      _ -> {:ok, nil}
```

Comment:
Same observation here, if something went wrong with the `file_server()`, we will be clueless.

Reply:
I think webhook failure reporting is out of scope for now, after deploying the feature, if we experience failure we could expose them in the webapp